### PR TITLE
refactor: align backend response status with UI

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

After changes introduced in #4346, the pending status represents exactly what the `missing` status is. This PR deprecates the `missing` status for record responses in favor of a new `pending` status.

@damianpumar, maybe you can adat the changes in the frontend part.

Refs: #4346 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [X] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,10 +22,14 @@ These are the section headers that we use:
 - Added `POST /api/v1/me/responses/bulk` endpoint to create responses in bulk for current user. ([#4380](https://github.com/argilla-io/argilla/pull/4380))
 - Added new CLI task to reindex datasets and records into the search engine. ([#4404](https://github.com/argilla-io/argilla/pull/4404))
 
-### Fixed
+###Fixed
```
Comment (frascuchon) [inline CHANGELOG.md]:
```suggestion
### Fixed
```

**Comment by jfcalvo** [comment]
Is not possible to do a data migration updating old `missing` to `pending` so we can remove the old code?

**Comment by damianpumar** [comment]
Hi Team! 
Yes, for sure is necessary to align the frontend with backend. @frascuchon do you need to do the change now? 
FYI: It would be no more than 15 minutes. Just let me know and add my commit here.

**Comment by damianpumar** [comment]
Hi guys! the CI is not working due to some failed tests, could you take a look?

**Comment by frascuchon** [comment]
> Is not possible to do a data migration updating old `missing` to `pending` so we can remove the old code?

The `missing` status is not stored in the DB, since reflects the status when a record has no response. So, there is no need for a migration definition. Anyway, we must keep backward compatibility with 2 versions behind, letting client users adapt to the new change, so old code cannot be removed with these changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ These are the section headers that we use:
 - Fixed error in `TextClassificationSettings.from_dict` method in which the `label_schema` created was a list of `dict` instead of a list of `str`. ([#4347](https://github.com/argilla-io/argilla/pull/4347))
 - Fixed total records on pagination component ([#4424](https://github.com/argilla-io/argilla/pull/4424))
 
+### Deprecated
+
+- The `missing` response status for filtering records is deprecated and will be removed in the release 1.23.0. Use `pending` instead. ([#4433](https://github.com/argilla-io/argilla/pull/4433))
```
Comment (frascuchon) [inline CHANGELOG.md]:
```suggestion
- The `missing` response status for filtering records is deprecated and will be removed in the release v1.24.0. Use `pending` instead. ([#4433](https://github.com/argilla-io/argilla/pull/4433))
```

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**